src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/components/step_index_pattern/step_index_pattern.js

```diff
@@ -77,6 +77,10 @@ export class StepIndexPattern extends Component {
 
     if (query.endsWith('*')) {
       const exactMatchedIndices = await ensureMinimumTime(getIndices(esService, query, MAX_SEARCH_SIZE));
+      // If the search changed, discard this state
+      if (query !== this.state.query) {
+        return;
```

Comment:
From React docs:

> setState() does not immediately mutate this.state but creates a pending state transition. Accessing this.state after calling this method can potentially return the existing value. There is no guarantee of synchronous operation of calls to setState and calls may be batched for performance gains.

So, I'm not sure that the this.state.query will always return the new query.  I think Chris D's suggestion about a queue and the ability to cancel inflight requests is the way to go.

Reply:
Yea that's a great point, but I don't think we need to build out a queue for this use case.

We can simply store an instance variable that represents the `lastQuery` and read from that instead of state. I'll push this up in a min.

Reply:
makes sense